src/protocol/bigreq.rs

```diff
@@ -99,6 +100,10 @@ impl TryParse for EnableReply {
             return Err(ParseError::ParseError);
         }
         let result = EnableReply { sequence, length, maximum_request_length };
+        let remaining = remaining.get((4 - (initial_value.len() - remaining.len()) % 4) % 4..)
+            .ok_or(ParseError::ParseError)?;
+        let remaining = remaining.get(32usize.saturating_sub(initial_value.len() - remaining.len())..)
+            .ok_or(ParseError::ParseError)?;
         Ok((result, remaining))
```

Comment:
I haven't understood this complicated dance yet, but assuming I understand the goal correctly... how about the following instead?
```rust
let parsed_length = initial_value.len() - remaining.len();
let actual_length = 32 + length * 4; // TODO: Did I remember this formula correctly?
let remaining = remaining.get((actual_length - parsed_length)..).ok_or(ParseError::ParseError)?;
```